src/zen/websocket_server.cpp

```diff
@@ -0,0 +1,1360 @@
+                LogPrint("ws", "%s():%d - invalid obj\n", __func__, __LINE__);
+                return INVALID_PARAMETER;
+            }
+            CBlockIndex* pblockindex = NULL;
+            uint256 hash(uint256S(o.get_str()));
+            {
+                LOCK(cs_main);
+                BlockMap::iterator mi = mapBlockIndex.find(hash);
+                if (mi != mapBlockIndex.end())
+                {
+                    pblockindex = (*mi).second;
+                    if (!chainActive.Contains(pblockindex))
+                        pblockindex = NULL;
+                }
+            }
+            if (pblockindex == NULL)
+            {
+                LogPrint("ws", "%s():%d - block index not found for hash[%s], skipping it\n", __func__, __LINE__, o.get_str());
+                continue;
+            }
+            if (pblockindex->nHeight > lastH)
+            {
+                lastH = pblockindex->nHeight;
+                pblockindexStart = pblockindex;
+            }
+        }
+        {
+            LOCK(cs_main);
+            CBlockIndex* nextBlock = pblockindexStart;
+            if (nextBlock == NULL)
+            {
+                LogPrint("ws", "%s():%d - start block index not found\n", __func__, __LINE__);
+                return INVALID_PARAMETER;
+            }
+            int n = 0;
+            while (n < len) {
+                listBlock.push_back(nextBlock);
+                nextBlock = chainActive.Next(nextBlock);
+                if (nextBlock == NULL)
+                    break;
+                n++;
+            }
+        }
+        sendHashes(listBlock.front()->nHeight, listBlock, WsEvent::MSG_RESPONSE, clientRequestId);
+        return OK;
+    }
+
+    int sendRawMempool(const std::string& clientRequestId) {
+        LOCK(mempool.cs);
+
+        std::list<uint256> hashes;
+        BOOST_FOREACH(const PAIRTYPE(uint256, CTxMemPoolEntry)& entry, mempool.mapTx)
+        {
+           hashes.push_back(entry.first);
+        }
+        sendMempool(hashes.size(), hashes, WsEvent::MSG_RESPONSE, clientRequestId);
+        return OK;
+
+    }
+
+    int sendMempoolTxs(const UniValue& hashes, const std::string& clientRequestId) {
+        LOCK(mempool.cs);
+
+        std::list<std::string> txs;
+        for(const UniValue& hash : hashes.getValues()) {
+            if (hash.isObject())
+            {
+                LogPrint("ws", "%s():%d - invalid obj\n", __func__, __LINE__);
+                return INVALID_PARAMETER;
+            }
+            uint256 txid(uint256S(hash.get_str()));
+            auto entry = mempool.mapTx.find(txid);
+            if (entry != mempool.mapTx.end()) {
+                txs.push_back(EncodeHexTx(
+                		static_cast<CTransaction>(entry->second.GetTx())));
+            } else {
+                LogPrint("ws", "%s():%d - mempool transaction not found!\n", txid.ToString(), __LINE__);
+            }
+
+        }
+        sendTxs(txs, WsEvent::MSG_RESPONSE, clientRequestId);
```

Comment:
We need because the explorer compares the transactions received the websocket with the transactions ids sent and it deletes the missing transactions in the response.